admin/deploy.yaml

```
@@ -30,10 +30,26 @@
           recipients: server-crew-qa@lists.canonical.com
     builders:
       - shell: |
-          #!/bin/bash -eux
-          rm -rf *
-          git clone https://github.com/canonical/server-jenkins-jobs
-          cd server-jenkins-jobs
+          #!/bin/bash
+
+          set -ux -o pipefail
+
+          # Setup a venv and install JJB, reuse the venv if already existing.
+          # Avoid leaving a broken venv behind is something goes wrong here.
+          trap 'rm -rf job-venv' EXIT
+          [[ -d job-venv ]] || python3 -m venv job-venv || exit 1
+          source job-venv/bin/activate
```

```suggestion
          . job-venv/bin/activate
```

`source` is an easily-avoidable bashism.